scylla-server-rust/src/controller/run_controller.rs

```diff
@@ -0,0 +1,62 @@
+use axum::{
+    extract::{Path, State},
+    Json,
+};
+use serde::Serialize;
+
+use crate::{error::ScyllaError, prisma, services::run_service, Database};
+
+#[derive(Serialize, Debug, PartialEq)]
+pub struct RunSend {
+    pub id: i32,
+    #[serde(rename = "locationName")]
+    pub location_name: String,
+    #[serde(rename = "driverName")]
+    pub driver_name: String,
+    #[serde(rename = "systemName")]
+    pub system_name: String,
+    pub time: i64,
+}
+
+impl From<&prisma::run::Data> for RunSend {
+    fn from(value: &prisma::run::Data) -> Self {
+        RunSend {
+            id: value.id,
+            location_name: value.location_name.clone().unwrap_or_default(),
+            driver_name: value.driver_name.clone().unwrap_or_default(),
+            system_name: value.system_name.clone().unwrap_or_default(),
+            time: value.time.timestamp_millis(),
+        }
+    }
+}
+
+pub async fn get_all_runs(State(db): State<Database>) -> Result<Json<Vec<RunSend>>, ScyllaError> {
+    let data = run_service::get_all_runs(&db).await?;
+
+    let transformed_data: Vec<RunSend> = data.iter().map(RunSend::from).collect();
+
+    Ok(Json::from(transformed_data))
+}
+
+pub async fn get_run_by_id(
+    State(db): State<Database>,
+    Path(run_id): Path<i32>,
+) -> Result<Json<RunSend>, ScyllaError> {
+    let data = run_service::get_run_by_id(&db, run_id).await?;
+
+    if data.is_none() {
+        return Err(ScyllaError::NotFound);
```

Comment:
this shouldnt be in the controller, we should be erroring inside the service

Reply:
I disagree on this, as I think the service doesnt care about the lack of existence of data, in my opinion it should just return what the database fetches.  Doing this in the controller allows the http-related stuff to be propogated independently of the service, so therefore unit tests and other callers could use the service functions without needing to handle http-related helpers.  Sorry I am probably over the top on this but I really like the idea of really simple and stupid services that only adhere to the prisma library standards.